src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -0,0 +1,1186 @@
+            }
+            Assert.Equal(1, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void TestGetFieldsToEmit_Initializer()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public string P { get => field; } = string.Empty;
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            for (int i = 0; i < 3; i++)
+            {
+                var fields = ((SourceMemberContainerTypeSymbol)comp.GetTypeByMetadataName("C")!).GetFieldsToEmit().ToArray();
+                Assert.Equal(1, fields.Length);
+                Assert.Equal("<P>k__BackingField", fields[0].Name);
+            }
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void AssignReadOnlyOnlyPropertyOutsideConstructor()
+        {
+            var comp = CreateCompilation(@"
+class Test
+{
+    int X
+    {
+        get
+        {
+            field = 3;
+            X = 3;
+            return 0;
+        }
+    }
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            comp.VerifyDiagnostics(
+                // PROTOTYPE(semi-auto-props): From review,
+                // This error doesn't make sense to me. I understand that the spec requires this field to be read-only, but I don't think this restriction is justified.
+                // (8,13): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)
+                //             field = 3;
+                Diagnostic(ErrorCode.ERR_AssgReadonly, "field").WithLocation(8, 13),
+                // (9,13): error CS0200: Property or indexer 'Test.X' cannot be assigned to -- it is read only
+                //             X = 3;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyProp, "X").WithArguments("Test.X").WithLocation(9, 13)
+            );
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact(Skip = "PROTOTYPE(semi-auto-props): Assigning in constructor is not yet supported.")]
+        public void AssignReadOnlyOnlyPropertyInConstructor()
+        {
+            var comp = CreateCompilation(@"
+using System;
+
+_ = new Test();
+
+class Test
+{
+    public Test()
+    {
+        X = 3;
+        Console.WriteLine(X);
+    }
+
+    int X
+    {
+        get { return field; }
+    }
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            comp.VerifyDiagnostics();
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
```

Comment:
>Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding); [](http://example.com/codeflow?start=12&length=70)

In general, if a test calls ```CompileAndVerify```, it would be more appropriate to assert this after the call. There could be exceptions where want to test specific situation and compare the count before and after the call, but I think this test doesn't fall into that category.

Reply:
Fixed